Adopt more smart pointers addressing UncountedLocalVarsChecker warning

rwlbuis opened the pull request:
#### aaea02d09e528cdd2284887ef407a86586380c07
<pre>
Adopt more smart pointers addressing UncountedLocalVarsChecker warning
<a href="https://bugs.webkit.org/show_bug.cgi?id=275562">https://bugs.webkit.org/show_bug.cgi?id=275562</a>

Reviewed by Ryosuke Niwa.

Adopt more smart pointers addressing the
[alpha.webkit.UncountedLocalVarsChecker] warning.

* Source/WebCore/css/CSSCounterStyleRegistry.cpp:
(WebCore::CSSCounterStyleRegistry::resolveExtendsReference):
(WebCore::CSSCounterStyleRegistry::counterStyle):
* Source/WebCore/css/ImmutableStyleProperties.cpp:
(WebCore::ImmutableStyleProperties::ImmutableStyleProperties):
* Source/WebCore/html/PublicURLManager.cpp:
(WebCore::PublicURLManager::revoke):
(WebCore::PublicURLManager::stop):
* Source/WebCore/loader/archive/ArchiveResourceCollection.cpp:
(WebCore::ArchiveResourceCollection::archiveResourceForURL):

Canonical link: <a href="https://commits.webkit.org/280112@main">https://commits.webkit.org/280112@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/70698fb681da54b6e86f811b6a49bb9222670c0b

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/55707 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/35030 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/8174 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/58691 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/6138 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/57833 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/42652 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/6336 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/44863 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/4224 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/57736 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/32931 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/48018 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/25996 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/29718 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/5347 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/4281 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/51696 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/5614 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/60282 "Built successfully") | 
| | [⏳ 🛠 vision ](https://ews-build.webkit.org/#/builders/visionOS-1-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/5745 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/52291 "Passed tests") | 
| | [⏳ 🛠 vision-sim ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-Build-EWS "Waiting in queue, processing has not started yet") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/48088 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/51784 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/12361 "Built successfully and passed tests") | [⏳ 🧪 vision-wk2 ](https://ews-build.webkit.org/#/builders/visionOS-1-Simulator-WK2-Tests-EWS "Waiting in queue, processing has not started yet") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/30861 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/31946 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/33027 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/31693 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Source/WebCore/css/ImmutableStyleProperties.cpp:
```diff
@@ -41,8 +41,8 @@ ImmutableStyleProperties::ImmutableStyleProperties(const CSSProperty* properties
         metadataArray[i] = properties[i].metadata();
-        auto* value = properties[i].value();
-        valueArray[i] = value;
+        auto value = properties[i].protectedValue();
```
I've heard people saying that in cases like this we should do this instead:
RefPtr value = properties[I].value();

rwlbuis replied:
Fixed.

achristensen07 commented on Source/WebCore/css/CSSCounterStyleRegistry.cpp:
```diff
@@ -135,13 +135,12 @@ RefPtr<CSSCounterStyle> CSSCounterStyleRegistry::counterStyle(const AtomString&
     if (map) {
-        auto counter = getCounter(name, *map);
-        if (counter)
+        if (auto counter = getCounter(name, *map))
```
I feel like we may as well change auto to RefPtr here to make it more clear to humans reading this that it is a smart pointer.

rwlbuis replied:
Yes, more readable indeed.

achristensen07 commented on Source/WebCore/loader/archive/ArchiveResourceCollection.cpp:
```diff
@@ -59,7 +59,7 @@ void ArchiveResourceCollection::addResource(Ref<ArchiveResource>&& resource)
 ArchiveResource* ArchiveResourceCollection::archiveResourceForURL(const URL& url)
 {
-    if (auto* resource = m_subresources.get(url.string()))
+    if (SUPPRESS_UNCOUNTED_LOCAL auto* resource = m_subresources.get(url.string()))
```
Why not just change auto* to RefPtr?

rwlbuis replied:
It is an extra ref, but this is not critical code AFAIK. Done.

rniwa commented on Source/WebCore/css/ImmutableStyleProperties.cpp:
```diff
@@ -41,8 +41,8 @@ ImmutableStyleProperties::ImmutableStyleProperties(const CSSProperty* properties
-        auto* value = properties[i].value();
-        valueArray[i] = value;
+        RefPtr value = properties[i].value();
+        valueArray[i] = value.get();
```
Rather than calling `get()` then manually calling `ref()`, we should call `leakRef()` instead.

rwlbuis replied:
Makes sense, done.

rwlbuis committed:
Adopt more smart pointers addressing UncountedLocalVarsChecker warning
https://bugs.webkit.org/show_bug.cgi?id=275562

Reviewed by Ryosuke Niwa.

Adopt more smart pointers addressing the
[alpha.webkit.UncountedLocalVarsChecker] warning.

* Source/WebCore/css/CSSCounterStyleRegistry.cpp:
(WebCore::CSSCounterStyleRegistry::resolveExtendsReference):
(WebCore::CSSCounterStyleRegistry::counterStyle):
* Source/WebCore/css/ImmutableStyleProperties.cpp:
(WebCore::ImmutableStyleProperties::ImmutableStyleProperties):
* Source/WebCore/html/PublicURLManager.cpp:
(WebCore::PublicURLManager::revoke):
(WebCore::PublicURLManager::stop):
* Source/WebCore/loader/archive/ArchiveResourceCollection.cpp:
(WebCore::ArchiveResourceCollection::archiveResourceForURL):

Canonical link: https://commits.webkit.org/280112@main
